tiingo/src/endpoint/crypto/prices.ts

```diff
@@ -0,0 +1,72 @@
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { ExecuteWithConfig, Config } from '@chainlink/types'
+
+export const NAME = 'prices'
+
+export interface ResponseSchema {
+  ticker: string
+  baseCurrency: string
+  quoteCurrency: string
+  priceData: {
+    date: string
+    open: number
+    high: number
+    low: number
+    close: number
+    volume: number
+    volumeNotional: number
+    fxOpen: number
+    fxHigh: number
+    fxLow: number
+    fxClose: number
+    fxVolumeNotional: number
+    fxRate: number
+    tradesDone: number
+  }[]
+}
+
+const customParams = {
+  base: ['base', 'from', 'coin'],
+  quote: ['quote', 'to', 'market'],
+  field: false,
+}
+
+// When an invalid symbol is given the response body is empty
+const customError = (data: ResponseSchema[]) => !data.length
+
+export const execute: ExecuteWithConfig<Config> = async (request, config) => {
+  const validator = new Validator(request, customParams)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const base = validator.validated.data.base.toLowerCase()
+  const quote = validator.validated.data.quote.toLowerCase()
+  const field = validator.validated.data.field || 'fxClose'
+  const url = '/tiingo/crypto/prices'
+
+  const options = {
+    ...config.api,
+    params: {
```

Comment:
do we need to do any deep merging here? or `config.api` is only top level objects?

Reply:
Won't need any deep merging here. 

Here is what `config.api` looks like:

```
api: {
      withCredentials: !!apiKey,
      baseURL: getEnv(ENV_API_ENDPOINT, prefix),
      baseWsURL: getEnv(ENV_WS_API_ENDPOINT, prefix),
      timeout: parseInt(timeout || '') || DEFAULT_API_TIMEOUT,
      headers: {
        common: {
          'Cache-Control': 'no-cache, no-store, must-revalidate',
          Pragma: 'no-cache',
          'Content-Type': 'application/json',
          Accept: 'application/json',
        },
      },
    },
```